pkg/runtime/init.go

```diff
@@ -0,0 +1,115 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package runtime
+
+import (
+	"context"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/golang/glog"
+
+	"istio.io/mixer/pkg/adapter"
+	cpb "istio.io/mixer/pkg/config/proto"
+	"istio.io/mixer/pkg/config/store"
+	"istio.io/mixer/pkg/expr"
+	"istio.io/mixer/pkg/pool"
+	"istio.io/mixer/pkg/template"
+)
+
+// This file contains code to create new objects that are
+// of package wide interest.
+
+// New creates a new runtime Dispatcher
+// Create a new controller and a dispatcher.
+// Returns a ready to use dispatcher.
+func New(eval expr.Evaluator, gp *pool.GoroutinePool, handlerPool *pool.GoroutinePool,
```

Comment:
The `New` method does return a dispatcher, but it does more than that. 
It creates a controller, creates a dispatcher, associates one with the other, and fetches initial configuration. 

In effect this is creating new `runtime`.